packages/core/src/lib/Decider.ts

```diff
@@ -101,6 +105,29 @@ export class Decider<Event, State> {
     )
   }
 
+  /**
+   * 1. Invoke the supplied `interpret` function with the present state to determine whether any write is to occur.
+   * 2. (if events yielded) Attempt to sync the yielded events to the stream.
+   *    (Restarts up to `maxAttempts` times with updated state per attempt, throwing `MaxResyncsExhaustedException` on failure of final attempt.)
+   * 3. Returns the new version of the stream
+   */
+  transactVersion(
```

Comment:
wondering whether I should rename my `TransactWithPostVersion` helpers - I guess this naming does align with the rest of the names....

Reply:
Given you're in a lang with function overloading and this is not, I'm not sure we should aim for function name parity. That said, `TransactWithPostVersion(decide)` is so long that us mere `fantomas` users might think twice before using it, so it's probably worth shortening it!

Reply:
Yes function name parity is definitely not a goal
I'm just thinking the burden of overly long names should probably fall on people doing more fiddly and.or questionable things like checking input versions, whreas returning the post version is relatively normal and sane.